spec/activerecord-bitemporal/transaction_at_spec.rb

```diff
@@ -33,4 +37,340 @@
+      # new transaction time     :                               |<---------->|
+      it_behaves_like "valid uniqueness" do
+        let(:new_from) { active2_to + 2.days }
+        let(:new_to) { new_from + 10.days }
+      end
+
+      # active1 transaction time : |<---------->|
+      # active2 transaction time :          |<---------->|
+      # new transaction time     :                    |<---------->|
+      it_behaves_like "invalid uniqueness" do
+        let(:new_from) { active1_to - 2.days }
+        let(:new_to) { active2_from + 2.days }
+      end
+
+      # active1 transaction time : |<---------->|
+      # active2 transaction time :              |<---------->|
+      # new transaction time     :                           |<---------->|
+      it_behaves_like "valid uniqueness" do
+        let(:new_from) { active2_to }
+        let(:new_to) { new_from + 10.days }
+      end
+    end
+
+    describe ".create" do
+      subject { -> { EmployeeWithUniquness.create!(name: "Tom") } }
+      context "exists destroyed model" do
+        let(:employee) { EmployeeWithUniquness.create!(name: "Jane").tap { |it| it.update(name: "Tom") } }
+        before do
+          employee.destroy!
+        end
+        it { is_expected.not_to raise_error }
+      end
+
+      context "exists past model" do
+        before do
+          EmployeeWithUniquness.create!(name: "Tom", created_at: "1982/12/02", deleted_at: "2001/03/24")
+        end
+        it { is_expected.not_to raise_error }
+      end
+
+      context "exists future model" do
+        before do
+          EmployeeWithUniquness.create!(name: "Tom", created_at: Time.current + 10.days)
+        end
+        it { is_expected.to raise_error ActiveRecord::RecordInvalid }
+      end
+    end
+
+    context "`valid_datetime` out of range `valid_from` ~ `valid_to`" do
+      it { is_expected.to be_truthy }
+
+      context "empty records" do
+        before { EmployeeWithUniquness.destroy_all }
+        subject { EmployeeWithUniquness.new(valid_from: "9999/1/10").valid_at("9999/1/1", &:save) }
+        it { is_expected.to be_truthy }
+      end
+    end
+
+    context "duplicated `valid_from` `valid_to` with non set `created_at`" do
+      let(:created_at) { "2019/04/01".to_time }
+      let(:deleted_at) { "2019/10/01".to_time }
+      let(:valid_from) { "2019/01/01".to_time }
+      let(:valid_to) { "2019/06/01".to_time }
+      before do
+        EmployeeWithUniquness.create!(created_at: created_at, deleted_at: deleted_at, valid_from: valid_from, valid_to: valid_to, name: "Jane")
+      end
+
+      context "current time in created_at ~ deleted_at" do
+        it do
+          Timecop.freeze(created_at + 1.days) {
+            emp = EmployeeWithUniquness.new(valid_from: valid_from, valid_to: valid_to, name: "Jane")
+            expect(emp).to be_invalid
+          }
+        end
+      end
+
+      context "current time in created_at ~ deleted_at" do
```

Comment:
I think this context is incorrect.
`current time greater than deleted_at`

Reply:
Thanks!
I fixed it.